### DISCUSSION TITLE
Timed out fetching a new connection

### DISCUSSION THREAD

**Discussion by chanlito** [prisma/prisma#15242] [Q&A] [answered]
![image](https://user-images.githubusercontent.com/15667259/189090010-a1fabbd6-bf86-44bd-98c7-78706aaea216.png)

![image](https://user-images.githubusercontent.com/15667259/189090509-2a127386-f624-4f2a-a784-9a1f6a7f9b4f.png)

Does anyone know what's causing this issues? our production application is running fine, and all of the sudden we start getting this error.

**Comment by janpio** [comment, marked as answer]
There is a configured timeout how long Prisma Client should wait for for a connection to be fetched from the connection pool.
The connection pool is configured to have a specific number of connections.
When all those connections are busy, a Client has to wait until one is free sometimes.
If it has to wait longer than the timeout, you get this error.

So you just have all connections busy all the time. That can be because if you have more traffic, slower queries, more queries. Solutions can be to fix all of that, or increase the timeout of the connection limit.

(Older versions of Prisma might also have bugs in this code - so please make sure you are on the newest version.)
